packages/core/src/mastra/index.ts

```diff
@@ -479,9 +479,61 @@ do:
 
     registerHook(AvailableHooks.ON_SCORER_RUN, createOnScorerHook(this));
 
+    /*
+      Register Mastra instance with AI tracing exporters and initialize them
+    */
+    if (config?.observability) {
+      this.registerAITracingExporters();
+      // Fire-and-forget initialization - individual errors are already logged
+      void this.initAITracingExporters();
+    }
+
     this.setLogger({ logger });
   }
 
+  /**
+   * Register this Mastra instance with AI tracing exporters that need it
+   */
+  private registerAITracingExporters(): void {
+    const allTracingInstances = getAllAITracing();
+    allTracingInstances.forEach(tracing => {
+      const exporters = tracing.getExporters();
+      exporters.forEach(exporter => {
+        // Check if exporter has __registerMastra method
+        if ('__registerMastra' in exporter && typeof (exporter as any).__registerMastra === 'function') {
```

Comment:
shouldn't we do instanceof a class instead? Or Can't typescript safe us? Or does not all exporters have this method?

Reply:
most exporters won't need an instance of mastra and won't have this method.  Only ones  like the default that need storage access will.